cmd/counters/src/lib.rs

```diff
@@ -259,6 +409,471 @@ fn counter_dump(
+                });
+            }
+            Order::Alpha => {
+                self.methods.sort_unstable_by(|a, _, b, _| a.cmp(b));
+            }
+        }
+    }
+}
+
+impl fmt::Display for IpcIface<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let total_len = f.width().unwrap_or(10);
+        let dim_space = " ".dimmed();
+        const INDENT: &str = "    ";
+        for (method_name, ctrs) in &self.methods {
+            let total = ctrs.total();
+            if total == 0 && !f.alternate() {
+                continue;
+            }
+            let total_str = total.to_string();
+            let errors = ctrs.total_errors();
+            let ok = total - errors;
+            let ok_str = ok.to_string();
+            let err_str = errors.to_string();
+            writeln!(
+                f,
+                " fn {}::{}() {dim_space:.>pad$}{} calls\n{INDENT}{}:",
+                self.name.bold(),
+                method_name.bold(),
+                total_str.bold(),
+                "clients".bold(),
+                pad = 80usize
+                    .saturating_sub(" fn ".len())
+                    .saturating_sub("() ".len())
+                    .saturating_sub("::".len())
+                    .saturating_sub(" calls".len())
+                    .saturating_sub(method_name.len())
+                    .saturating_sub(self.name.len())
+                    .saturating_sub(total_str.len())
+                    .saturating_sub(1)
+            )?;
+            let mut formatted_tasks = 0;
+            let mut formatted_errors = 0;
+            let num_important_tasks = ctrs
+                .0
+                .iter()
+                .filter(|(_, c)| c.total() > 0 || f.alternate())
+                .count();
+            for ((task, gen), ctrs) in &ctrs.0 {
+                let total = ctrs.total();
+                if total == 0 && !f.alternate() {
+                    continue;
+                } else {
+                    formatted_tasks += 1;
+                }
+
+                let errors = ipc_error_count(ctrs);
+                let ok = total - errors;
+                let ok_str = format!(
+                    "{} {ok}",
+                    if num_important_tasks > 1 { "+" } else { "=" }
+                );
+                let restarts = match gen {
+                    GenOrRestartCount::Gen(gen) => {
+                        format!(" (gen {gen:?})")
+                    }
+                    GenOrRestartCount::RestartCount(restarts) => {
+                        format!(" ({restarts} restarts)")
+                    }
+                };
+                write!(f, "{INDENT}task {}{restarts} ", task.italic())?;
+                if errors == 0 && !f.alternate() {
+                    let err_str =
+                        if num_important_tasks > 1 { "+ 0" } else { "= 0" };
+                    let pad1 = 80usize
+                        .saturating_sub(err_str.len())
```

Comment:
All these are _almost_ an argument for using `Saturating` here, buuuut I think it'd wind up just as verbose and I'm not sure.

Reply:
TIL about `Saturating`! That must be new!

Reply:
Oh, it is --- stabilized in 1.74.0!

I think these would be a bit less verbose using `Saturating`, since we could just wrap the `80` in `Saturating` and then do the rest of the arithmetic normally, but it would require updating our MSRV to 1.74.0...